app/src/main/java/com/android/voyageur/ui/components/SearchBar.kt

```diff
@@ -0,0 +1,57 @@
+package com.android.voyageur.ui.components
+
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Search
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+/**
+ * Reusable search bar component with consistent styling and search icon.
+ *
+ * @param placeholderId Resource ID for placeholder text
+ * @param onQueryChange Callback triggered when search query changes
+ * @param modifier Optional modifiers for the search bar
+ */
```

Comment:
For the search bar I think it could be a good idea to include a clear button (an X) at the end of the bar. There is something similar implemented for the location fields in add trip/activity. Let me know your thoughts.

Reply:
Thank you for the suggestion! I think it would enrich the UX